### PR TITLE
Add hvp helpers

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

**Comment by wsmoses** [comment]
Fixes https://github.com/EnzymeAD/Enzyme.jl/issues/1205

**Comment by wsmoses** [comment]
@vchuravy after a few attempts I can't seem to get the doctest filter syntax right if you know the right incantation

File: Project.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 CEnum = "0.4, 0.5"
 ChainRulesCore = "1"
 EnzymeCore = "0.7.5"
-Enzyme_jll = "0.0.127"
+Enzyme_jll = "0.0.126"
```
Comment (giordano) [inline Project.toml:23]: Why is this necessary?

Comment (wsmoses) [reply Project.toml:23]: oh CI wasn't running at all without the other package registered, so i did this temporarily to see if it would run

Comment (giordano) [reply Project.toml:23]: Maybe there was a delay in the registry?  [v0.0.127](https://github.com/JuliaRegistries/General/pull/109889) was registered over 24 hours ago.
